### PR TITLE
Added exporting contacts

### PR DESCRIPTION
**Pull request by AbdussamadA**
Export contacts just like labels

### CODE REVIEW

**Comment by AbdussamadA** [comment]
this is for #3904

File: gui/qt/contact_list.py [hunk 1]
```diff
@@ -33,7 +33,7 @@
 from PyQt5.QtWidgets import (
     QAbstractItemView, QFileDialog, QMenu, QTreeWidgetItem)
 from .util import MyTreeWidget
-
+import os, json
```
Comment (bauerj) [inline gui/qt/contact_list.py]: Please follow [PEP8](https://www.python.org/dev/peps/pep-0008/#imports) for the import order:

> Imports should be grouped in the following order:
>
> 1. standard library imports
> 2. related third party imports
> 3. local application/library specific imports

So this should be moved up to `import webbrowser`.

Comment (AbdussamadA) [reply gui/qt/contact_list.py]: done

File: lib/contacts.py [hunk 2]
```diff
@@ -63,6 +64,14 @@ def import_file(self, path):
         self.update(d)
         self.save()
 
+    def export_file(self, fileName):
+        try:
+            with open(fileName, 'w+') as f:
+                json.dump(self, f, indent=4, sort_keys=True)
+        except (IOError, os.error) as reason:
+            traceback.print_exc(file=sys.stderr)
+            raise reason
```
Comment (SomberNight) [inline lib/contacts.py]: Not sure how much sense it makes to catch these exception and just re-raise them like this, here.
Could you maybe do it similar to `import_file`? Define `FileExportFailed` and raise that here.

File: lib/contacts.py [hunk 3]
```diff
@@ -63,6 +64,14 @@ def import_file(self, path):
         self.update(d)
         self.save()
 
+    def export_file(self, fileName):
+        try:
+            with open(fileName, 'w+') as f:
+                json.dump(self, f, indent=4, sort_keys=True)
```
Comment (SomberNight) [inline lib/contacts.py]: btw you can add a "success" pop-up as you originally did, I think that's a good idea. A clean way to do it is with a `try-except-else`, in the `else` branch. If you do, add that to that importing too. :P

File: gui/qt/contact_list.py [hunk 4]
```diff
@@ -63,12 +64,22 @@ def import_contacts(self):
...
         self.on_update()
 
+    def export_contacts(self):
+        filename = self.parent.getSaveFileName(_("Select file to save your contacts"), 'electrum_contacts.json', "*.json")
+        if  not filename:
+            return
+        try:
+            self.parent.contacts.export_file(filename)
+        except (IOError, os.error) as reason:
+            self.parent.show_critical(_("Electrum was unable to export your contacts.") + "\n" + str(reason))
```
Comment (SomberNight) [inline gui/qt/contact_list.py]: btw you can add a "success" pop-up as you originally did, I think that's a good idea. A clean way to do it is with a `try-except-else`, in the `else` branch. If you do, add that to that importing too. :P
(deleted the other comment regarding this, it was at the wrong pos.)

File: lib/contacts.py [hunk 5]
```diff
@@ -54,15 +55,23 @@ def import_file(self, path):
         try:
             with open(path, 'r') as f:
                 d = self._validate(json.loads(f.read()))
-        except json.decoder.JSONDecodeError:
+        except ValueError:
```
Comment (SomberNight) [inline lib/contacts.py]: What's the reason for this change?

File: lib/contacts.py [hunk 6]
```diff
@@ -54,15 +55,23 @@ def import_file(self, path):
...
             traceback.print_exc(file=sys.stderr)
-            raise FileImportFailedEncrypted()
+            raise FileImportFailedInvalidJSON()
```
Comment (SomberNight) [inline lib/contacts.py]: You are right, the file might not be encrypted, all we know is that JSON decoding failed. `FileImportFailedEncrypted` was just a guess. However I don't think we should start introducing all kinds of  `File*Failed*` exceptions. 
How about keeping only `FileImportFailed` and `FileExportFailed`, and extending `FileImportFailed` with a message the same way you have done for `FileExportFailed`?

File: lib/contacts.py (hunk 5 above)
Comment (AbdussamadA) [reply lib/contacts.py]: https://stackoverflow.com/a/44714607/672396

File: lib/contacts.py (hunk 6 above)
Comment (AbdussamadA) [reply lib/contacts.py]: ok will do. also shall i do this same treatment for invoices? invoices are payment requests right?

Comment (AbdussamadA) [reply lib/contacts.py]: yes they are the same as payment requests. i'll do invoice exporting as well.

Comment (SomberNight) [reply lib/contacts.py]: Invoices are what you should pay; i.e. the stuff at the bottom of the `Send` tab (when visible).

File: lib/contacts.py (hunk 5 above)
Comment (SomberNight) [reply lib/contacts.py]: Ok. Can you add a comment that reflects this? like `# backwards compatibility for JSONDecodeError ` or similar
